lib/system.nim

```diff
@@ -2503,12 +2503,13 @@ proc `==`*[T](x, y: seq[T]): bool {.noSideEffect.} =
     when not defined(JS):
       proc seqToPtr[T](x: seq[T]): pointer {.inline, nosideeffect.} =
         result = cast[pointer](x)
-    else:
```

Comment:
This entire diff looks suspicous to me.

Reply:
This just checks if both params are backed by the same object. Since unsafeAddr can't be called here (seqs aren't treated as mapped directly to objects, so you'd a [box] to get the (base, index) ptr), I can't see an alternative way to do this check here.